### PR TITLE
feat: verify client's public key in auth

### PR DESCRIPTION
**Pull request by kdonthi**
* Match public key to player ID in custom auth that wraps `EthereumWalletAuth`

### CODE REVIEW

File: client/src/core/utils.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+export function getPlayer(): PlayerSchema | undefined {
+    const game = getGame()
+    if (game === undefined) {
+        return undefined
+    }
+
+    const playerTag = playerIdTag(game.GameId)
+    const playerIDStr = localStorage.getItem(playerTag);
```
Comment (kdonthi) [inline client/src/core/utils.ts:31]: use getPlayerIDTag here

File: server/startup/initGame.go [hunk 2]
```diff
@@ -33,11 +33,11 @@ func InitWorld(ctx *server.EngineCtx) {
 		}
 	}
 
-	// add admin player
-	adminPlayerId := -100
-	createPlayerRequest := systems.CreatePlayerRequest{
-		PlayerID: adminPlayerId,
-	}
+	//// add admin player
+	//adminPlayerId := -100
```
Comment (kdonthi) [inline server/startup/initGame.go:37]: remove

File: server/systems/createPlayerSystem.go [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
+	"fmt"
 	"github.com/curio-research/keystone-starter-kit/data"
 	"github.com/curio-research/keystone-starter-kit/helper"
 	"github.com/curio-research/keystone/server"
 )
 
 type CreatePlayerRequest struct {
-	PlayerID int `json:"playerId"`
+	PlayerID        int    `json:"playerID"`
+	Base64PublicKey string `json:"publicKey"`
```
Comment (kdonthi) [inline server/systems/createPlayerSystem.go:12]: change the json to match?

File: server/systems/verifyAuth.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	"encoding/json"
+	"github.com/curio-research/keystone/server"
+)
+
+const PlayerIDHeader = "playerIDHeader"
+
+const PlayerIDTag = "playerID"
+const PublicKeyTag = "publicKey"
+
+func VerifyIdentity[T any]() server.IMiddleware[T] {
```
Comment (kdonthi) [inline server/systems/verifyAuth.go:13]: VerifyWalletAndIdentity

File: client/src/core/middleware/ethereumWalletAuth.ts [hunk 5]
```diff
@@ -13,23 +12,27 @@ interface EthereumWalletAuth {
 
 // Middleware function to validate
 // Data payload is signed with an ethereum wallet
+
+export const ECDSAPublicKeyAuthHeader = "ecdsaPublicKeyAuth"
```
Comment (kevinz917) [inline client/src/core/middleware/ethereumWalletAuth.ts:16]: I think we can rename all things ECDSA to eth related to be consistent

File: client/src/keystone/stateStore.ts [hunk 6]
```diff
@@ -90,6 +91,7 @@ export class WorldState {
 
     this.applyAllPendingUpdates();
     this.setIsFetchingState(false);
+    createPlayer();
```
Comment (kevinz917) [inline client/src/keystone/stateStore.ts:94]: the stateStore is part of the Keystone folder which is kind of part of the SDK so i think we should instantiate this outside. StateStore should be handling state, only

File: client/src/pages/Game.tsx [hunk 7]
```diff
@@ -1,25 +1,32 @@
...
+import {uiState} from 'index';
 import Resources from 'components/Resources';
 
+import {getPlayerID} from "../core/utils";
+
 export const toast = createStandaloneToast();
 
-// hard coded playerID. See InitGame.go
+// hard coded playerID. See InitGame.go TODO use random # generator
 const playerId = -100;
```
Comment (kevinz917) [inline client/src/pages/Game.tsx:18]: Unused

File: client/src/pages/Game.tsx [hunk 8]
```diff
@@ -42,7 +49,7 @@ const Game = observer(() => {
         break;
 
       case ' ':
-        const lastPressedDirection = uiState.lastMovedDirection;
+        const lastPressedDirection = uiState.lastMovedDirection; // TODO so you need to move before firing?
```
Comment (kevinz917) [inline client/src/pages/Game.tsx:52]: Assuming yes, because without this you don't know where to fire

File: server/data/schemas.go [hunk 9]
```diff
@@ -38,10 +39,11 @@ type TileSchema struct {
...
 type PlayerSchema struct {
-	Id        int       `gorm:"primaryKey;autoIncrement:false"`
-	Position  state.Pos `gorm:"embedded"`
-	Resources int
-	PlayerId  int
+	Id              int       `gorm:"primaryKey;autoIncrement:false"`
+	Position        state.Pos `gorm:"embedded"`
+	Resources       int
+	PlayerId        int
+	Base64PublicKey string
```
Comment (kevinz917) [inline server/data/schemas.go:46]: Rename to EthPublicKey

File: server/systems/createPlayerSystem.go [hunk 10]
```diff
@@ -1,36 +1,44 @@
...
+	"fmt"
 	"github.com/curio-research/keystone-starter-kit/data"
 	"github.com/curio-research/keystone-starter-kit/helper"
 	"github.com/curio-research/keystone/server"
 )
 
 type CreatePlayerRequest struct {
-	PlayerID int `json:"playerId"`
+	PlayerID        int    `json:"playerId"`
+	Base64PublicKey string `json:"base64PublicKey"`
```
Comment (kevinz917) [inline server/systems/createPlayerSystem.go:12]: Rename to EthKey (avoid using base64 language in all scenarios except internals. devs are familiar with Eth language)

**Review by kevinz917 — APPROVED** [review]

**Pull request merged**